server/src/test/java/org/elasticsearch/index/IndexVersionTests.java

```diff
@@ -199,23 +199,20 @@ public void testParseLenient() {
 
     public void testLuceneVersionOnUnknownVersions() {
         // between two known versions, should use the lucene version of the previous version
-        IndexVersion version = IndexVersionUtils.getPreviousVersion();
-        final IndexVersion nextVersion = IndexVersion.fromId(version.id() + 100);
-        if (IndexVersionUtils.allReleasedVersions().contains(nextVersion) == false) {
-            // the version is not known, we make an assumption the Lucene version stays the same
-            assertThat(version.luceneVersion(), equalTo(nextVersion.luceneVersion()));
-        } else {
-            // the version is known, the most we can assert is that the Lucene version is not earlier
-            // Version does not implement Comparable :(
-            assertTrue(nextVersion.luceneVersion().onOrAfter(version.luceneVersion()));
-        }
+        IndexVersion previousVersion = IndexVersionUtils.getPreviousVersion();
+        IndexVersion currentVersion = IndexVersion.current();
+        int intermediateVersionId = previousVersion.id() + randomInt(currentVersion.id() - previousVersion.id() - 1);
```

Comment:
How does this handle the previous version being immediately next to the current version (eg 8,500,051 and 8,500,052)?

Reply:
it will create an intermediate version that is equal to the previous one in that case

Reply:
In that case the assert below will fail - as it will be a released version

Reply:
Woops, yes indeed!

Reply:
I'll just remove this check then and update the comment.